runtime/compiler/runtime/RuntimeAssumptions.hpp

```diff
@@ -197,6 +198,7 @@ class TR_PersistentClassInfo : public TR_Link0<TR_PersistentClassInfo>
    flags8_t                            _flags;
    flags8_t                            _shouldNotBeNewlyExtended; // one bit for each possible compilation thread
    CCVResult                           _ccvResult;
+   TR_PersistentMemory                *_persistentMemory;
```

Comment:
Adding an extra pointer to `TR_PersistentClassInfo` will increase footprint because we have such an entry for every class. It is the reason we only allow 8 compilation threads (`_shouldNotBeNewlyExtended` has a bit for every compilation thread and we decided to limit that at 8)

Reply:
My suggestion was to add an optional TR_Memory parameter to `removeSubClasses()` so that it knows exactly what persistent memory to use for deallocation. If the parameter is not passed, then it can use the global persistent memory. Interestingly, I don't see who calls `removeSubClasses()` for non-jitserver case.

Reply:
I see, I didn't think think footprint would be a concern here.
I'll modify it to do what you suggest.

Reply:
This is outside of the scope of this PR, but I think is worth noting:

If the footprint of this class is indeed a concern (8 bytes per class doesn't seem like a lot...), I think it can actually be reduced. The `_nameLength` field doesn't seem to be used anywhere, and removing it would reduce the size by 8 bytes (same as would be added by the `_persistentMemory` pointer). The last 3 fields could be reordered to allow extending `_shouldNotBeNewlyExtended` to `flags16_t`, which would then enable up to 16 compilation threads instead of 8 (assuming this is the only place that causes the limit). `_ccvResult` could also be encoded in the `_flags` if we ever need another 8 bits.

Reply:
Indeed. I didn't realize that `_nameLength` is no longer used. I have no recollection for what could have been used.
Maybe will do that in another PR.